Edit the Machine ID docs refactor

ptgott opened the pull request:
- **Rename the new guides:** Use the "Connect a Bot" and "Deploy Machine ID" language instead of "Access Guides" and "Platform Guides" to connect these guides more explicitly to the language we use in the "Concepts" discussion of the Machine ID landing page.

- **Add context to the deployment guide index page**: Reduce repetition and provide information about each deployment method to help users get more context about how Machine ID runs and joins a cluster, as well as to help users choose a deployment guide.

- **Make links more visible on the Machine ID intro page:** Use a video banner for the Machine ID intro so it takes up less space on the page. Shorten some sections and add more specific H2s for the links.

- **Streamline some deployment guides:** Where guides include "Background" and "Guide" H2s, blend the introductory information with the guide so we can promote the "Step" H2s to H3s and direct the reader to the step-by-step instructions more quickly.

- **Add new pages to the docs table of contents.**

ptgott committed:
Edit the Machine ID docs refactor

- **Rename the new guides:** Use the "Connect a Bot" and "Deploy Machine
  ID" language instead of "Access Guides" and "Platform Guides" to
  connect these guides more explicitly to the language we use in the
  "Concepts" discussion of the Machine ID landing page.

- **Add context to the deployment guide index page**: Reduce repetition
  and provide information about each deployment method to help users get
  more context about how Machine ID runs and joins a cluster, as well as
  to help users choose a deployment guide.

- **Make links more visible on the Machine ID intro page:** Use a video
  banner for the Machine ID intro so it takes up less space on the page.
  Shorten some sections and add more specific H2s for the links.

- **Streamline some deployment guides:** Where guides include
  "Background" and "Guide" H2s, blend the introductory information with
  the guide so we can promote the "Step" H2s to H3s and direct the reader
  to the step-by-step instructions more quickly.

- **Add new pages to the docs table of contents.**

zmb3 commented on docs/config.json:
```diff
@@ -1349,12 +1349,80 @@
+              "slug": "/machine-id/deployment/azure/"
+            },
+            {
+              "title": "CircleCI",
```
Any particular reason for this ordering?

I would put the public cloud providers together (Google Cloud, AWS, Azure), and then list the CI/CD platforms together (Jenkins, GHA, CircleCI, GitLab CI)

zmb3 commented on docs/pages/machine-id/deployment.mdx:
```diff
@@ -0,0 +1,71 @@
+  demonstrating to the Auth Service that it belongs in your Teleport cluster.
+- **Platform-signed document:** The platform that hosts `tbot`, such as a
+  Kubernetes cluster or Amazon EC2 instance, provides a signed identity document
+  that Teleport can verify using the platform's certificate authority. 
```
Should we emphasize that this is the preferred approach, as it's more secure (no shared secrets) and doesn't suffer from the one-time-use issues associated with static tokens?

ptgott replied:
Done. To keep this brief within the format of the list, I only mentioned the security consideration.

zmb3 commented on docs/pages/machine-id/introduction.mdx:
```diff
@@ -58,13 +40,12 @@ Bots share a number of similarities with human users:
-typically an identity issued to the machine by the platform it runs on (e.g. AWS
-IAM). This process allows the machine to authenticate with the Teleport
-cluster without using a long-lived secret.
+A bot authenticates to Teleport with a **join token**, which prevents it from
```
A join token can itself be a long-lived secret though. I kind of liked the "a bot does not log in" terminology.

ptgott committed:
Respond to zmb3 feedback

- List cloud platforms before CI/CD platforms on the sidebar
- Recommend using platform-signed identity documents in the deployment
  guide intro page.
- Edit language introducing join tokens.

strideynet commented on docs/pages/machine-id/deployment.mdx:
```diff
@@ -0,0 +1,72 @@
+
+The `tbot` binary runs as a container or on a Linux virtual machine. If you run
+`tbot` on GitHub Actions, you can use a ready-made [GitHub Actions
+workflow](https://github.com/marketplace/actions/teleport-auth).
```
We might want to avoid linking to a specific action here - since this one only works for SSH and we have a few others.

strideynet reviewed (approved):
Looks great - thanks. Just four tweaks :)

ptgott committed:
Respond to strideynet feedback

- Edit wording in the deployment guide index page, including renaming a
  section heading and adding language re: renewable certs in the static
  token join method.
- Change GitHub Actions link.
- Rename the Access Guides back to "Access Guides"
